src/alloc/Cargo.toml

```diff
@@ -33,3 +33,8 @@ jemalloc = ["tikv-jemallocator", "mz-prof", "mz-prof-http/jemalloc"]
 # all the Materialize binaries correctly serve heap profiling tools at the
 # `/prof` endpoints.
 normal = ["workspace-hack", "mz-prof-http"]
+
+[package.metadata.cargo-gazelle]
+# We manually write this BUILD.bazel file because we need to carefully
+# configure when `jemalloc` gets used for ASAN.
+skip_generating = true
```

Comment:
Does this mean `bin/bazel gen` won't work for these crates and we have to manually edit the bazel files when dependencies change? Might be worth pointing that out in `bazel.md`?

Reply:
In cargo builds, we avoid this issue by passing `--no-default-features`. Does an equivalent exist for bazel? If yes, we might not need the manual patch to the build files.

Reply:
Unfortunately a similar feature does not exist, the way you customize the build is by introducing a custom `config_setting` like I did here